433: Added if checks around class definitions.

toolstack opened the pull request:
This is to avoid conflicts with other plugins that may include locales.php for their own use.

Alternate if syntax used to avoid triggering coding standards review of entire class.

Resolves #413.

toolstack committed:
Added if checks around class definitions.

This is to avoid conflicts with other plugins that may include locales.php for their own use.

Alternate if syntax used to avoid triggering coding standards review of entire class.

toolstack committed:
Added newline.

ocean90 commented:
The path to locales.php can be changed via https://github.com/GlotPress/GlotPress-WP/blob/2.0.0/gp-settings.php#L4 which means that custom locales.php need to have the class_exists() too.
What do you think about adding a class_exists() check to https://github.com/GlotPress/GlotPress-WP/blob/2.0.0/gp-settings.php#L57?

toolstack commented:
I considered it but thought it was better inside the class file as if another plugin pulls a copy of it use then we make sure it doesn't break.

We could add the calls_exists() to both places but I didn't like the code duplication.

toolstack committed:
Add check when including the locales.php file for classes.

This makes sure that if a custom locales.php file has been used and doesn't have the required class checks, it won't break things.
